### PR TITLE
[.Net] rename Autogen.Ollama to AutoGen.Ollama and add more test cases to AutoGen.Ollama

### PR DESCRIPTION
**Pull request by LittleLittleCloud**
## Related issue number
#2512 

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [ ] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [ ] I've made sure all auto checks have passed.

### CODE REVIEW

File: dotnet/src/AutoGen.Ollama/Agent/OllamaAgent.cs [hunk 1]
```diff
@@ -69,22 +70,28 @@ public class OllamaAgent : IStreamingAgent
...
-                ChatResponse? chatMessage = JsonSerializer.Deserialize<ChatResponse>(line);
-                if (chatMessage == null) continue;
-                yield return new MessageEnvelope<ChatResponse>(chatMessage, from: Name);
+                    yield return new MessageEnvelope<ChatResponse>(finalUpdate, from: Name);
+                }
             }
         }
     }
+
+    public string Name { get; }
```
Comment (DavidLuong98) [inline dotnet/src/AutoGen.Ollama/Agent/OllamaAgent.cs:93]: Nit : Can we move Name up on line 24-26 with the other fields?

File: dotnet/.editorconfig [hunk 2]
```diff
@@ -141,7 +141,7 @@ csharp_preserve_single_line_statements = true
 csharp_preserve_single_line_blocks = true
 
 # Code block
-csharp_prefer_braces = false:none
+csharp_prefer_braces = true:warning
```
Comment (DavidLuong98) [inline dotnet/.editorconfig:144]: :sob:

File: dotnet/src/AutoGen.Ollama/Agent/OllamaAgent.cs [hunk 3]
```diff
@@ -69,22 +70,28 @@ public class OllamaAgent : IStreamingAgent
...
 
                 ChatResponseUpdate? update = JsonSerializer.Deserialize<ChatResponseUpdate>(line);
-                if (update != null)
+                if (update is { Done: false })
                 {
                     yield return new MessageEnvelope<ChatResponseUpdate>(update, from: Name);
                 }
+                else
+                {
+                    var finalUpdate = JsonSerializer.Deserialize<ChatResponse>(line) ?? throw new Exception("Failed to deserialize response");
```
Comment (DavidLuong98) [inline dotnet/src/AutoGen.Ollama/Agent/OllamaAgent.cs:85]: Should we use the async overload?

File: dotnet/src/AutoGen.Ollama/Agent/OllamaAgent.cs (hunk 1 above)
Comment (LittleLittleCloud) [reply dotnet/src/AutoGen.Ollama/Agent/OllamaAgent.cs:93]: The name is a property

File: dotnet/src/AutoGen.Ollama/Agent/OllamaAgent.cs (hunk 3 above)
Comment (LittleLittleCloud) [reply dotnet/src/AutoGen.Ollama/Agent/OllamaAgent.cs:85]: Not here, the async overload works with stream, not string

**Review by DavidLuong98 — APPROVED** [review]

File: dotnet/src/AutoGen.Ollama/Agent/OllamaAgent.cs (hunk 3 above)
Comment (DavidLuong98) [reply dotnet/src/AutoGen.Ollama/Agent/OllamaAgent.cs:85]: You are right :)

**Pull request merged**